src/modules/reviewrecords.js

```diff
@@ -38,16 +40,28 @@ class ReviewRecords {
         const updatedMultihash = await this.getLastReviewRecordUpdate(db, multihash);
         if (updatedMultihash) {
             // TODO: Check that the update is valid first
-            notifyUpdate(multihash, updatedMultihash);
+            try {
+                const reviewRecord = await this.readReviewRecord(updatedMultihash);
+                notifyUpdate(multihash, updatedMultihash, reviewRecord);
+            } catch (error) {
+                this.chluIpfs.logger.error('Review update ' + updatedMultihash + ' for ' + multihash + ' was invalid: ' + error);
```

Comment:
Should there be a call back to notifyUpdate with an error? What will happen if the the UI is waiting for this callback to be called?

Also, remind me again why we don't use a Promise here? I think you explained that to me already. Sorry to ask you to repeat yourself.

Reply:
No problem :+1: 

notifyUpdate is not really a callback, it's more like an handler of events that could be called multiple times or never at all. It is supposed to be called when the library discovers an update for a review (which could be never if there are no updates).

We did not use a promise because since orbit-db is _eventually consistent_ you only know if you are downloading updates right now but you can't know if there are updates to be downloaded (until you are downloading them). So how long do you wait while checking for updates? a few seconds, many seconds, a few ms? Whatever decision would not work for the majority of cases, so the best way I found to deal with this is that the readReviewRecord call immediately reads the specified version of the review, then if there are any updates discovered at any point in the future, it calls the notifyUpdate function that was passed by the readReviewRecord caller.

In the UI part, you call readReviewRecord and wait for the promise to resolve, then you know you have the RR you asked for. Then, if you receive a call to the notifyUpdate function you passed, you know you have to update the reviewRecord. in the Demo we used this to show updates by having the notifyUpdate function dispatch a redux action with the update information, then the reducer handles updating if the review record that was updated is relevant to the current context, otherwise it's just discarded.

The update checking logic has no polling, it works by listening to orbit-db events so it should be as lightweight as possible.

There are some optimizations we can make (for example if the review update database is already locally replicated we can immediately check and return an updated RR) but I did not want to overengineer since this system works already well in the demo

Reply:
OK. This is interesting. I wonder if passing a callback will result in the variables defined in the closure sticking around and not causing other surprises. 

Maybe we can just listen to an event from orbitdb and react to it, instead of passing a closure around? We don't need to fix this for this PR. It is just something to be aware of.